service/src/main/java/datawave/microservice/model/ModelController.java

```diff
@@ -0,0 +1,277 @@
+     * @HTTP 204 model not found
+     * @HTTP 500 internal server error
+     */
+    @Operation(summary = "Copy a model.")
+    
+    @PostMapping("/clone")
+    @Secured({"Administrator", "JBossAdministrator"})
+    public VoidResponse cloneModel(@RequestParam String name, @RequestParam String newName,
+                    @RequestParam(defaultValue = DEFAULT_MODEL_TABLE_NAME) String modelTableName, @AuthenticationPrincipal DatawaveUserDetails currentUser) {
+        VoidResponse response = new VoidResponse();
+        
+        Model model = getModel(name, modelTableName, currentUser);
+        // Set the new name
+        model.setName(newName);
+        insertMapping(model, modelTableName, currentUser);
+        return response;
+    }
+    
+    /**
+     * Retrieve the model and all of its mappings
+     *
+     * @param name
+     *            model name
+     * @param modelTableName
+     *            name of the table that contains the model
+     * @return the Model
+     * @RequestHeader X-ProxiedEntitiesChain use when proxying request for user
+     *            
+     * @HTTP 200 success
+     * @HTTP 404 model not found
+     * @HTTP 500 internal server error
+     */
+    @GetMapping("/{name}")
+    public Model getModel(@RequestParam String name, @RequestParam(defaultValue = DEFAULT_MODEL_TABLE_NAME) String modelTableName,
+                    @AuthenticationPrincipal DatawaveUserDetails currentUser) {
+        Model response = new Model(jqueryUri, dataTablesUri);
+        List<Key> keys;
+        try {
+            keys = accumloConnectionService.getKeys(modelTableName, currentUser, name);
+        } catch (TableNotFoundException e) {
+            QueryException qe = new QueryException(DatawaveErrorCode.MODEL_NAME_LIST_ERROR, e);
+            log.error(qe.getMessage());
+            response.addException(qe.getBottomQueryException());
+            return response;
+        }
+        
+        Set<Authorizations> auths = accumloConnectionService.getConnection(modelTableName, name, currentUser).getAuths();
+        TreeSet<FieldMapping> fields = response.getFields();
+        keys.forEach(k -> fields.add(ModelKeyParser.parseKey(k, auths)));
+        response.setName(name);
+        return response;
+    }
+    
+    /**
+     * <strong>Administrator credentials required.</strong> Insert a new field mapping into an existing model
+     *
+     * @param model
+     *            list of new field mappings to insert
+     * @param modelTableName
+     *            name of the table that contains the model
+     * @return a VoidResponse
+     * @RequestHeader X-ProxiedEntitiesChain use when proxying request for user
+     *            
+     * @HTTP 200 success
+     * @HTTP 500 internal server error
+     */
+    @Operation(summary = "Insert a new field mapping into an existing model.")
+    @PostMapping(value = {"/insert", "/import"}) // If we get to change to standard REST, this would just be / and rely on the
```

Comment:
unfinished comment

Reply:
im not sure what was meant by that comment.  deleted.